### PR TITLE
psuedo class first-child error

### PR DESCRIPTION
**Pull request by ankush29**

### CODE REVIEW

File: docs/examples/AsyncCallbacks.js [hunk 1]
```diff
@@ -19,10 +19,11 @@ const loadOptions = (inputValue, callback) => {
   }, 1000);
 };
 
-export default class WithCallbacks extends Component<*, State> {
+export default class WithCallbacks extends Component<State> {
   state = { inputValue: '' };
   handleInputChange = (newValue: string) => {
-    const inputValue = newValue.replace(/\W/g, '');
+    //const inputValue = newValue.replace(/\W/g, '');
```
Comment (csantos1113) [inline docs/examples/AsyncCallbacks.js]: Remove?

File: docs/markdown/renderer.js [hunk 2]
```diff
@@ -97,7 +97,7 @@ const Heading = props => {
   }
   const css = {
     marginTop: 0,
-    '&:not(:first-child)': { marginTop: 30 },
+    '&:not(:first-of-type)': { marginTop: 30 },
```
Comment (csantos1113) [inline docs/markdown/renderer.js]: Is there an open issue we could link for this fix?

File: docs/examples/AsyncCallbacks.js (hunk 1 above)
Comment (ankush29) [reply docs/examples/AsyncCallbacks.js]: for a special character, so we can enter them same as in promise

File: docs/markdown/renderer.js (hunk 2 above)
Comment (ankush29) [reply docs/markdown/renderer.js]: I guess no. I found it in the console panel

**Comment by serkangunes** [comment]
When is this going to be merged?

**Comment by wootwoot1234** [comment]
@JedWatson can we merge this?

**Comment by JedWatson** [comment]
Hey @ankush29, thanks for the PR!

This fix was also included in #2284 which has been merged now

tbh it's much easier to review and merge PRs that deal with a single isolated issue, so I really appreciate your contributions here, but it would be helpful for the other things you included if you can open separate PRs explaining the rationale for each - otherwise it's hard to know what you meant to change and why, and merge them individually.

Thanks again 🙂

**Comment by pserrer** [comment]
@JedWatson this PR also included code to focus the selected option, which doesn't seems to be part of #2284 or any other previous commit. I would be great to have this - or at least something else working, also included.

**Comment by Tirzono** [comment]
@ankush29 would you be willing to make a separate PR for the focus the selected option issue? I would be willing to do that as well, but don't want to steal the credits for it.

**Comment by ankush29** [comment]
Sure! I will do.

**Comment by ankush29** [comment]
Hey,

I have created a separate pull request :
https://github.com/JedWatson/react-select/pull/3813/commits/a67c5484454a086c152a436057b4c327833abf10

regards

**Comment by KyorCode** [comment]
@JedWatson any update on this PR?

**Comment by presiyan-kostov** [comment]
Any news about the pending PR here?

**Comment by Tirzono** [comment]
This PR is closed and won't be merged. It's split out over other PR's. Have a look at #3868 for example. Unfortunately the maintainers didn't have time yet to have a look at that one, so we just patiently have to wait.

**Pull request closed without merging**